internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -356,35 +356,38 @@ func (r *PreprovisioningImageReconciler) mapInfraEnvPPI() func(a client.Object)
 		reconcileRequests := []reconcile.Request{}
 
 		for i := range images {
-			// Don't queue if the Image URL and InfraEnv's URL is the same.
-			if images[i].Status.ImageUrl != infraEnv.Status.ISODownloadURL {
-				reconcileRequests = append(reconcileRequests, reconcile.Request{
-					NamespacedName: types.NamespacedName{
-						Namespace: images[i].Namespace,
-						Name:      images[i].Name,
-					},
-				})
-			}
+			reconcileRequests = append(reconcileRequests, reconcile.Request{
```

Comment:
This seems like an overkill and I think we can optimize it, not sure if it's worth the trouble though.
(we care about the imageURL, irnoicAgentImageOverride and the internalIgnitionOverride everything else is just noise).

Reply:
But we don't know the current state of those overrides. That's why we have to render the entire ignition to compare in the reconcile step.

It was easy with the url because we could compare them directly, but it's not so simple now.